### DISCUSSION TITLE
Thermostat over climate

### DISCUSSION THREAD

**Discussion by MBRetrofit** [jmcollin78/versatile_thermostat#201] [Ideas]
I am new and a little far away from the standard language in this community, so please accept my sorry in advance.

In the case of a "climate", I imagine that every "climate" has already his own Temperature sensor entity. 

In my case I want to build a Versatile Thermostat in each room ( in Living room I have 2 electric radiators and I want to use the Power Management feature between those 2) and all this Versatile's to group again into a Main Versatile Thermostate "over climate" for using the Power Management for entire house.

1 - Why on Thermostat over climate the "Temperature sensor entity id* " is still necessary as each climate already has it's own temp sensors.

2 - I read French and English information about Power management and I am still confused about this;

"Gives the power and max power sensor of your home.
Then specify the power consumption of the heater when on."

I imagine the algo over your algo like that:

1 - The sum of power of radiators ( live from a entity that sum's the each instantaneos consumtion)
2 - The total house power readed live from the main fuse
3 - The Max power fix value - contract value

Subtracting the 1 from 2 can be added to 1 and to be safeguarded to the 3.

Anyway, this Versatile seams the most modern and advanced solution here.

**Comment by wiktor-z** [comment]
While some climate devices, like TRVs, may come with built-in sensors, their readings can be biased due to their proximity to the heat source. Therefore, it's often beneficial to include a new sensor in the setup, positioned away from the heat source. This additional sensor can measure the ambient temperature in the desired location, providing a more accurate reference for setting the thermostat to the desired temperature.

> Anyway, this Versatile seams the most modern and advanced solution here.
I back this opinion ❤️ .

**Comment by MBRetrofit** [comment]
Great ideea, so as we use radiators ( or TRV'S) that already had internal temp sensor and we prefer to create each room climate for using external temp sensor and especialy for the possibilty to set independent temperature values in each room, let's try this:

1 - Let's add multiple Temperature sensors id - for each room - or NONE if the each room climate already has it
2 - Add the difference from Total House Power and Total Heater's Power.
3 - A new Card with all Climate's independetly displayed.

**Comment by MBRetrofit** [comment]
Is it such a bad ideea to eliminate the Temperature sensor entity ID* from Main mandatory attributes when setting a new Vtherm Thermostat over a climate? 

One of the best feature of this great VTherm is the energy management. Why not use this TPI between rooms, not only between heaters under the same temperature sensor...Syncronized triggering between rooms ...that's my dream.

**Reply by jmcollin78** [reply]
> Is it such a bad ideea to eliminate the Temperature sensor entity ID* from Main mandatory attributes when setting a new Vtherm Thermostat over a climate?

The main idea was to have the real room temperature and not the internal temperature which mostly wrong or biaised as said by @wiktor-z. So this is the main use case and you are, till now, the only one that think it is not always a good idea. If you want to give it a try, you can always create a virtual sensor that have its value given by the climate internal temperature, and give this sensor to the VTherm.

**Reply by maia** [reply]
There is one use case for *additionally* using the TRV internal temperature sensor and not only the room temperature sensor. Imagine the following:
- target temperature: 22°C
- room temperature: 20°C
- TRV temperature: increasing, now at 24°C

The TRV will turn the valve off, as the TRV temperature is above the target temperature. We now have to wait for two things to happen: a slow decrease of the TRV internal temperature (as the valve is closed) and a slow increase in regulated target temperature to something that is higher than the TRV temperature – because only then the TRV will open the valve again.

I'm not sure how to use the TRV temperature algorithmically, but I cannot say it's never of any use.

**Reply by MBRetrofit** [reply]
Maybe the internal temperature sensors can be used in safety scenarios.

I adjusted the internal temperature sensors to 26 or 28, so they will do nearly nothing.

**Comment by MBRetrofit** [comment]
Sorry for my english, 

My case, small house with 4 rooms more than 90% sunny days per year, superb Mediterranean climate in the south-east of Spain. I define human thermal confort as it is taught in higher education institutions and Google responds exactly the same way, very different from the new fashionable ineptitudes. 

"Humans generally feel comfortable between temperatures of 22 °C to 27 °C and a relative humidity of 40% to 60%."

The mighty sun has a very predictive cycle and give me the opportunity to install an electric heater in each room ( even 2 in Livingroom).
I choose "heat accumulator radiators" ( has some type of stone inside..)  for using the inertia as a helper for Vtherm algo.

So, every room has a zigbee switch connected to the electric heater and a zigbee temperature sensor somwhere far away from radiator to prevent bias. 
Every room use own Vtherm ( over switch and with external temperature sensor ) with TPI and Power Management, Windows managemnt, etc.
I do not consider internal temperature sensor from radiator.

I am very happy, but I strongly belive that I can integrate all four Over_Switch Vtherm's into one Over_Climate for my entire house.

Of course, my house does not turn arround over the sun, and the southern rooms will always need less power and the Northen side will need more.

And of course my energy contract is limited to 5,7KW and because I am a strong beliver in any smart algorithm's, I will succed to split this limit of power in slices exactly as the TPI do for my each room.

My wife cooks ...all electric, my son has very big power source at his PC...

I work in Automotive and all modern cars use a smart PWM management to feed power to the lights one by one ( for example). There is never 2 lights powered in the same time.  Just rotate the power with different speed to meet requirments.

4 Rooms, 4 Switch's connected to electric heaters, 4 separate Temperature sensors...each room independently and with different requests for the desire temperature easy and wonderful managed by Vtherm.

I only have to convince you to delete the temperature sensor id ( interior and exterior )  from Over_Climate ( of course I can create a dumy entity...) and most of all I have to convince you to syncronise the TPI over 4 rooms...Kind of TPI in TPI, or TPI over TPI...

At least in my case the Off time from first room ( priority room) is enough to include the next 3 On time from the rest of the rooms.

Thank you,

Alex
